public/expressions/overlay_anomalies.ts

```diff
@@ -0,0 +1,193 @@
+}
+
+const name = 'overlay_anomalies';
+
+export type OverlayAnomaliesExpressionFunctionDefinition =
+  ExpressionFunctionDefinition<'overlay_anomalies', Input, Arguments, Output>;
+
+// This gets all the needed anomalies for the given detector ID and time range
+const getAnomalies = async (
+  detectorId: string,
+  startTime: number,
+  endTime: number
+): Promise<AnomalyData[]> => {
+  const anomalySummaryQuery = getAnomalySummaryQuery(
+    startTime,
+    endTime,
+    detectorId,
+    undefined,
+    false
+  );
+
+  // We set the http client in the plugin.ts setup() fn. We pull it in here to make a
+  // server-side call directly.
+  // Note we can't use the redux fns here (e.g., searchResults()) since it requires
+  // hooks (e.g., useDispatch()) which doesn't make sense in this context, plus is not allowed by React.
+  const anomalySummaryResponse = await getClient().post(
+    `..${AD_NODE_API.DETECTOR}/results/_search`,
+    {
+      body: JSON.stringify(anomalySummaryQuery),
+    }
+  );
+
+  return parsePureAnomalies(anomalySummaryResponse);
+};
+
+// This takes anomalies and returns them as vis layer of type PointInTimeEvents
+const convertAnomaliesToLayer = (
+  anomalies: AnomalyData[],
+  ADPluginResource: PluginResource
+): PointInTimeEventsVisLayer => {
+  const events = anomalies.map((anomaly: AnomalyData) => {
+    return {
+      timestamp: anomaly.startTime + (anomaly.endTime - anomaly.startTime) / 2,
+      metadata: {},
+    };
+  });
+  return {
+    originPlugin: 'anomaly-detection',
+    type: VisLayerTypes.PointInTimeEvents,
+    pluginResource: ADPluginResource,
+    events: events,
+  } as PointInTimeEventsVisLayer;
+};
+
+export const overlayAnomaliesFunction =
+  (): OverlayAnomaliesExpressionFunctionDefinition => ({
+    name,
+    type: 'vis_layers',
+    inputTypes: ['vis_layers'],
+    help: i18n.translate('data.functions.overlay_anomalies.help', {
+      defaultMessage: 'Add an anomaly vis layer',
+    }),
+    args: {
+      detectorId: {
+        types: ['string'],
+        default: '""',
+        help: '',
+      },
+    },
+
+    async fn(input, args, context): Promise<ExprVisLayers> {
+      // Parsing all of the args & input
+      const detectorId = get(args, 'detectorId', '');
+      const timeRange = get(
+        context,
+        'searchContext.timeRange',
+        ''
+      ) as TimeRange;
+      const origVisLayers = get(input, 'layers', [] as VisLayers) as VisLayers;
+      const parsedTimeRange = timeRange ? calculateBounds(timeRange) : null;
+      const startTimeInMillis = parsedTimeRange?.min?.unix()
+        ? parsedTimeRange?.min?.unix() * 1000
+        : undefined;
+      const endTimeInMillis = parsedTimeRange?.max?.unix()
+        ? parsedTimeRange?.max?.unix() * 1000
+        : undefined;
+      const ADPluginResource = {
+        type: 'anomaly-detection-type',
+        id: detectorId,
+        name: PLUGIN_NAME,
+        urlPath: `${PLUGIN_NAME}#/detectors`,
+      };
```

Comment:
A few notes on this part:
- `type` will be the header used when categorizing and grouping the `VisLayer`s on the view events flyout, so ideally it should be something like `Anomaly Detectors` (this could be pulled into a constant as well)
- `urlPath` should be the full path to the relevant details page for the plugin resource - in our case, `${PLUGIN_NAME}#/detectors/${detectorId}/results` probably

I'll follow up by adding some more descriptions on these fields in the interface definitions. It might be worth adding some comments wherever these are implemented here too to prevent confusion.

Reply:
It's difficult to formalize this across plugins without limiting flexibility, since all we care about here is a string. I think we can leave this up to the plugins for now, as well as document the use of it in OSD. If that's still causing problems down the road we could revisit the interfaces.

Reply:
I am all for documenting as I understand the limitation around formalizing a specific standard. Good call adding some description in the interface definition. I will add some comment in this code and potentially we can also add more info in documentation for future plugin onboarding

Reply:
Forgot to also add:
`name` should be the name of the plugin resource, not the name of the plugin name. In this case, detector name

Reply:
added this